### PR TITLE
smol.py commits

### PR DESCRIPTION
**Pull request by swyxio**
this pr contains a working implementation of the smol.py agent adapter for rift

image

however it does not:

- create files as expected (should check for existence of files)

 hoping for @cristianoc to take it from here

## instructions

to run:

- `export OPENAI_API_KEY=___`
- `python -m rift.server.core --port 7797 --debug True` in one terminal
- `python rift/rift-engine/rift/llm/test.py`

### CODE REVIEW

File: rift-engine/rift/agents/code_completion.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ class CodeCompletionProgress(AgentProgress):
 
 @dataclass
 class CodeCompletionAgentParams:
-    task: str
+    instructionPrompt: str
```
Comment (swyxio) [inline rift-engine/rift/agents/code_completion.py:26]: rename agreed with @jesse

**Comment by cristianoc** [comment]
@swyxio added commit to make file changes work.
@jesse-michael-han comment on source of truth:
There's a question of what is the source of truth for file contents.
    The current assumption is that the source of truth is whatever is stored on disk (as opposed to: taking into account files that are currently being modified in some open window in the editor which executes agent instructions).

**Comment by jesse-michael-han** [comment]
> The current assumption is that the source of truth is whatever is stored on disk (as opposed to: taking into account files that are currently being modified, and unsaved, in some open window in the editor which executes agent instructions).

Yeah this is good for now. It is lower friction than using the lower git commit and avoids the complications of considering other AI-generated edits in unsaved workspace buffers.

**Comment by jesse-michael-han** [comment]
awesome work y'all

**Pull request merged**